lib/Semantics/check-do-forall.cpp

```diff
@@ -912,7 +912,7 @@ void DoForallChecker::CheckForBadLeave(
 static bool StmtMatchesConstruct(const parser::Name *stmtName,
     StmtType stmtType, const parser::Name *constructName,
     const ConstructNode &construct) {
-  bool inDoConstruct{MaybeGetDoConstruct(construct)};
+  bool inDoConstruct{static_cast<bool>(MaybeGetDoConstruct(construct))};
```

Comment:
I think this is clearer:
`bool inDoConstruct{MaybeGetDoConstruct(construct) != nullptr};`

Reply:
Replaced with your suggestion